storage/google/cloud/storage/notification.py

```diff
@@ -82,6 +90,40 @@ def __init__(self, bucket, topic_name,
         if payload_format is not None:
             self._properties['payload_format'] = payload_format
 
+    @classmethod
+    def from_api_repr(cls, resource, bucket):
+        """Construct an instance from the JSON repr returned by the server.
+
+        See: https://cloud.google.com/storage/docs/json_api/v1/notifications
+
+        :type resource: dict
+        :param resource: JSON repr of the notification
+
+        :type bucket: :class:`google.cloud.storage.bucket.Bucket`
+        :param bucket: Bucket to which the notification is bound.
+
+        :rtype: :class:`BucketNotification`
+        :returns: the new notification instance
+        :raises ValueError:
+            if resource is missing 'topic' key, or if it is not formatted
+            per documented spec.
```

Comment:
We should help our users here and explain what our interpretation of "document spec" is.

Reply:
998b9f9